vpc-sample/main.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2020 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file holds functions for testing VPC ingress/egress with Cloud Functions
+
+# [START egress_hello_world]
+import os
+import urllib
+
+
+def get_hello_world(request):
+    try:
+        url = os.environ.get("URL")
+        req = urllib.request.Request(url)
+        # [END egress_hello_world]
+
+        ## Auth is only required for internal testing
```

Comment:
It complicates the documentation / tutorial, which should be focusing on the network settings.  I think it should be kept as simple as possible so people don't conflate authentication and network settings.  

That being said, I can be convinced otherwise.

Reply:
I agree that it adds complexity.

My point here is relying on the ingress protection alone is "perimeter security", but our general philosophy and recommendations on security are about defense in depth. (https://cloud.google.com/beyondcorp) One value of this is the ability to differentiate legitimate requests from inside the perimeter from requests entering through a security breach.

The way I approach this is that how-to's should be very focused on the topic at hand, but tutorials should aim for minimum production safety.

I think for a tutorial sample, it's worth the complexity and we can take pains to point out the two layers of security involved.

Reply:
I totally understand the concern about added complexity and loss of strict focus on network settings. That said, given our ultimate goal of supporting real use cases, I think it makes sense to include authentication. I can revise the accompanying tutorial to mention the two layers of security.

We need to determine if this changes any steps in the tutorial though. Do we want to rely on the default IAM configuration (which I'd then want to explain in the tutorial), or do we want to explicitly give the calling function or service an identity ([GCF](https://cloud.google.com/functions/docs/securing/function-identity), [Cloud Run](https://cloud.google.com/run/docs/securing/service-identity)) and then give that identity "invoker" permissions on the restricted function?

Reply:
SGTM - I've added the authentication back into the sample and added the commands for setting up the service account and invoking permissions into my tutorial doc at the bottom.

Reply:
Following up in email, since I think this might get lost in PR wrangling.